tokens/figma/utils.mjs

```diff
@@ -0,0 +1,335 @@
+    };
+
+    const existingCollections =
+      figmaData.meta.variableCollections;
+
+    function generateTempId(name) {
+      return `tempId_${name}`;
+    }
+
+    function updateCollection(
+      collectionName,
+      existingCollections
+    ) {
+      // Find the existing collection by name
+      const existingCollection = Object.values(
+        existingCollections
+      ).find(
+        (collection) =>
+          collection.name === collectionName
+      );
+
+      if (existingCollection) {
+        // If the collection exists, update it
+        newData.variableCollections.push({
+          action: "UPDATE",
+          id: existingCollection.id,
+          name: collectionName,
+        });
+      } else {
+        // If the collection doesn't exist, create it
+        const tempId = generateTempId(
+          collectionName
+        );
+        newData.variableCollections.push({
+          action: "CREATE",
+          id: tempId,
+          name: collectionName,
+        });
+      }
+    }
+
+    // Process each collection name
+    collections.forEach((collection) => {
+      updateCollection(
+        collection,
+        existingCollections
+      );
+    });
+
+    // Return the processed data for further use
+    return newData;
+  } catch (error) {
+    console.error("Error:", error);
+    throw error; // rethrow the error to be handled later
+  }
+}
+
+export async function updateOrCreateMode({
+  mode,
+  defaultModeName,
+  targetCollectionName,
+  existingCollections,
+}) {
+  const collection = Object.values(
+    existingCollections
+  ).find(
+    (collection) =>
+      collection.name === targetCollectionName
+  );
+
+  if (!collection) {
+    console.warn(
+      `Collection ${targetCollectionName} not found.`
+    );
+    return;
+  }
+
+  const collectionId = collection.id;
+  const existingModes = collection.modes || [];
+
+  // Find the default mode (e.g., "Mode 1" or "Default")
+  const defaultMode = existingModes.find(
+    (m) => m.name === DMODE_NAMES.DEFAULT // Replace with actual default mode name if different
+  );
+
+  // Find the target mode by its name
+  const existingMode = existingModes.find(
+    (m) => m.name === mode.name
+  );
+
+  if (mode.name === defaultModeName) {
+    if (defaultMode) {
+      // Rename the default mode to the target mode name
+      return {
+        action: "UPDATE",
+        id: defaultMode.modeId,
+        name: mode.name, // Rename "Default" to the target mode name
+        variableCollectionId: collectionId,
+      };
+    } else {
+      // If default mode does not exist, create it
+      return {
+        action: "CREATE",
+        id: generateTempModeId(
+          mode.name, // Use mode name for temp ID
+          targetCollectionName
+        ),
+        name: mode.name, // Create the mode with the target name
+        variableCollectionId: collectionId,
+      };
+    }
+  } else if (!existingMode) {
+    // If mode doesn't exist, create it
```

Comment:
this code is a copy of the else block from above. I think all this logic can be simplified

Reply:
I've changed the entire function